src/instructlab/cli/data/list.py

```diff
@@ -39,5 +65,34 @@ def list_datasets(dataset_dirs):
         click.secho(f"Failed to list datasets with exception: {exc}")
         raise click.exceptions.Exit(1)
 
+    grouped_data = defaultdict(list)
+    timestamp_to_model = {}
+    general_files = []
+
+    # Process model name files e.g. start with test_, train_, messages_
+    for item in data:
+        filename = item[0]
+        group_key = extract_model_and_timestamp(filename)
+
+        if group_key:
+            if "General" in group_key:
```

Comment:
What if General was in the model name?  One way you could avoid this string check is to return a tuple specifying whether the match was the general case.